core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1869,8 +1869,7 @@ public void _run() throws IOException, InstallationStatus {
                 // Do this first so we can avoid duplicate downloads, too
                 // check to see if the plugin is already installed at the same version and skip it
                 LOGGER.info("Skipping duplicate install of: " + plugin.getDisplayName() + "@" + plugin.version);
-                //throw new Skipped(); // TODO set skipped once we have a status indicator for it
-                return;
+                throw new Skipped();
```

Comment:
I do not see any value in that as well. I will play with it a bit more to get ti hidden and not spit exceptions (is situation that are not entirely clear to me).